### PR TITLE
protocol_subtomo_subtraction: improved implementation with MPI and md convert

### PR DESCRIPTION
**Pull request by EstrellaFG**
This PR goes with https://github.com/I2PC/xmipp/pull/825

The implementation has changed: now it converts the set of subtomograms to a xmd, subtract all of them (doing all the preprocessing needed in xmipp) and then it returns the output xmd which is converted to a setOfSubtomograms.

This development has implied to create a xmipptomo convert (based on xmipp3 convert) with some adaptation to subtomograms and coordinates3D. Probably this convert can be improved in the future.

### CODE REVIEW

**Review by Vilax — APPROVED** [review]

File: xmipptomo/protocols/protocol_subtraction_subtomo.py [hunk 1]
```diff
@@ -77,87 +75,29 @@ def _defineParams(self, form):
                       expertLevel=LEVEL_ADVANCED, help='Save input volume 1 (first subtomogram of the set) filtered '
                                                        'and input volume 2 (average) adjusted, which are the volumes '
                                                        'that are really subtracted.')
+        form.addParallelSection(threads=0, mpi=4)
```
Comment (albertmena) [inline xmipptomo/protocols/protocol_subtraction_subtomo.py:78]: Why 4 and not 8? (is the default in other cases)

Comment (albertmena) [reply xmipptomo/protocols/protocol_subtraction_subtomo.py:78]: Why we can not fill the MPI number in the form?
image

Comment (Vilax) [reply xmipptomo/protocols/protocol_subtraction_subtomo.py:78]: Despite I am still learning the algorithm the MPI can overload the RAM. That means at least 8 copies in memory, but the algorithm can requires many more (internaly)

Comment (EstrellaFG) [reply xmipptomo/protocols/protocol_subtraction_subtomo.py:78]: Yes, and AFAIK 4 was the maximum allowed in the machine where Scipion tests were executed (maybe this has changed now, not sure). But I have no problem if you change it as you consider.

With respect to why we can not fill the MPI number in the form, I could when I tried (actually I tried with different values), so not sure what has changed (maybe something from scipion is not updated or has changed?). I am so sorry but I can not check it now as my Ubuntu has died (I do not know why... the destiny?) but let me know if you need something @albertmena .

**Pull request merged**